--- a/src/ui/menus.ui
+++ b/src/ui/menus.ui
@@ -7,10 +7,6 @@
         <attribute name="label" translatable="yes" context="Action">Open</attribute>
         <attribute name="action">win.view-selection</attribute>
       </item>
-      <item>
-        <attribute name="label" translatable="yes">_Open With…</attribute>
-        <attribute name="action">win.open-with</attribute>
-      </item>
     </section>
     <section>
       <item>
